setup/ecr/inputs.tf

```diff
@@ -0,0 +1,36 @@
+
+variable "name_prefix" {
+  type        = string
+  description = "The prefix to prepend to resource names"
+
+  validation {
+    condition     = can(regex("^[[:alnum:]\\-]+$", var.name_prefix))
+    error_message = "name_prefix can only contain letters, numbers, and hyphens"
+  }
+}
+
+variable "name" {
+  type        = string
+  description = "The name to give to give to this ALB and its related resources"
+
+  validation {
+    condition     = can(regex("^[[:alnum:]\\-]+$", var.name))
+    error_message = "name can only contain letters, numbers, and hyphens"
+  }
+}
+
+variable "scan_images" {
+  type        = bool
+  default     = false
+  description = "Whether to scan images pushed to the ECR repo"
+}
+
+variable "source_account" {
```

Comment:
Chris, do you mind giving an example of when we will want to use a source_account that's different from the current account? (and why is this only a need for ECR?)

Reply:
In most situations you would have a solid separation between different components using AWS accounts as the security boundary.  It's not uncommon to have one team set up permissions/repos/etc in one central account that can be locked down, with other accounts owned by different teams (for running services, for example) using those centrally-managed resources.  You would often even have separate accounts for prod and non-prod services, maybe even one per environment (dev, staging, test, qa, prod, alpha, beta, gamma, etc).  For our immediate use case for Doorway we're just going to deploy everything altogether in one single account, but adding a `source_account` variable means we aren't forcing that design pattern on anyone.

Reply:
As for why only needed for ECR, that's likely not the only place we'll use it.  It's just the only resource we have in the setup project right now.

Reply:
thanks for the explanation!